lib/document/mongoDbRepository/DocumentMongoDbRepository.js

```diff
@@ -288,10 +288,10 @@ class DocumentMongoDbRepository {
     protocolVersion,
     data,
   }) {
-    const encodedProperties = this.dataContract.getEncodedProperties(this.documentType);
+    const binaryProperties = this.dataContract.getBinaryProperties(this.documentType);
```

Comment:
You can pass `promoteBuffers: true` option to `find` methods to get back native Buffers so you won't need to do this conversion.